frontend/app/src/main/java/com/quickpick/MyRecyclerView.java

```diff
@@ -0,0 +1,83 @@
+package com.quickpick;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.EditText;
+
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.quickpick.payloads.IdeaPayload;
+
+import java.util.List;
+
+class MyRecyclerViewAdapter extends RecyclerView.Adapter<MyRecyclerViewAdapter.ViewHolder> {
+
+    private List<IdeaPayload> mData;
+    private LayoutInflater mInflater;
+    private ItemClickListener mClickListener;
+
+    // data is passed into the constructor
+    MyRecyclerViewAdapter(Context context, List<IdeaPayload> data) {
+        this.mInflater = LayoutInflater.from(context);
+        this.mData = data;
+    }
+
+    // inflates the row layout from xml when needed
+    @Override
+    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view = mInflater.inflate(R.layout.recycler_view_entries, parent, false);
+        return new ViewHolder(view);
+    }
+
+    // binds the data to the TextView in each row
+    @Override
+    public void onBindViewHolder(ViewHolder holder, int position) {
+        String entry = mData.get(position).getName();
+        String description = mData.get(position).getDescription();
+        holder.myEntry.setText(entry);
+        holder.myEntry.setText(description);
+    }
+
+    // total number of rows
+    @Override
+    public int getItemCount() {
+        return mData.size();
+    }
+
+
+    // stores and recycles views as they are scrolled off screen
+    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
+        EditText myEntry;
```

Comment:
Prefer making these private and use getter/setters instead of exposing these externally